server/internal/adapter/gql/resolver_mutation_field.go

```diff
@@ -68,18 +67,25 @@ func (r *mutationResolver) UpdateField(ctx context.Context, input gqlmodel.Updat
 		return nil, err
 	}
 
-	ms, err := usecases(ctx).Model.FindByIDs(ctx, []id.ModelID{mId}, getOperator(ctx))
-	if err != nil || len(ms) != 1 || ms[0].ID() != mId {
-		if err == nil {
-			return nil, rerror.NewE(i18n.T("not found"))
-		}
+	m, err := usecases(ctx).Model.FindByID(ctx, mId, getOperator(ctx))
+	if err != nil {
 		return nil, err
 	}
-	m := ms[0]
 
-	s, err := usecases(ctx).Schema.FindByID(ctx, m.Schema(), getOperator(ctx))
-	if err != nil {
-		return nil, err
+	var s *schema.Schema
+	if input.Metadata != nil && *input.Metadata {
+		if m.Metadata() == nil {
+			return nil, rerror.NewE(i18n.T("metadata schema not found"))
+		}
+		s, err = usecases(ctx).Schema.FindByID(ctx, *m.Metadata(), getOperator(ctx))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		s, err = usecases(ctx).Schema.FindByID(ctx, m.Schema(), getOperator(ctx))
+		if err != nil {
+			return nil, err
+		}
 	}
```

Comment:
These logic should be moved to the usecase layer. This is the adapter layer, so it should be thin.